utils/roi_tools.py

```diff
@@ -0,0 +1,374 @@
+"""
+ This module represents functionality used by the `run.py` script for readability
+ and encapsulation.  References to `prior art` are given where appropriate.
+ Areas for future implementation are noted for ease of prospective implementation.
+"""
+
+from collections import OrderedDict
+import logging
+import numpy as np
+import os.path as op
+from skimage import draw
+
+log = logging.getLogger(__name__)
+
+
+class InvalidDICOMFile(Exception):
+    """Exception raised when an invalid DICOM file is encountered.
+
+    Attributes:
+        expression -- input expression in which the error occurred
+        message -- explanation of the error
+    """
+
+    def __init__(self, message):
+        Exception.__init__(self)
+        self.message = message
+
+
+class InvalidConversion(Exception):
+    """Exception raised when a conversion cannot be done (nifti to dicom, nifti to nrrd).
+
+    Attributes:
+        expression -- input expression in which the error occurred
+        message -- explanation of the error
+    """
+
+    def __init__(self, message):
+        Exception.__init__(self)
+        self.message = message
+
+
+class InvalidROIError(Exception):
+    """Exception raised when session ROI data is invalid.
+
+    Attributes:
+        expression -- input expression in which the error occurred
+        message -- explanation of the error
+    """
+
+    def __init__(self, message):
+        Exception.__init__(self)
+        self.message = message
+
+
+def poly2mask(vertex_row_coords, vertex_col_coords, shape):
```

Comment:
Not going to lie, I don't think I have the expertise in OHIF ROIs, or linear algebra to adequately review these utilities.